cosmovisor/cmd/cosmovisor/cmd/version.go

```diff
@@ -1,36 +1,89 @@
 package cmd
 
 import (
+	"encoding/json"
 	"fmt"
 	"os"
+	"strings"
 	"time"
 
+	"github.com/cosmos/cosmos-sdk/cosmovisor"
 	cverrors "github.com/cosmos/cosmos-sdk/cosmovisor/errors"
 	"github.com/rs/zerolog"
 )
 
-// Version represents Cosmovisor version value. Overwritten during build
-var Version = "1.1.0"
-
-// VersionArgs is the strings that indicate a cosmovisor version command.
-var VersionArgs = []string{"version", "--version"}
+var (
+	// FlagJSON formats the output in json
+	FlagJSON = "--json"
+	// Version represents Cosmovisor version value. Overwritten during build
+	Version = "1.1.0"
+	// VersionArgs is the strings that indicate a cosmovisor version command.
+	VersionArgs = []string{"version", "--version"}
+)
 
 // IsVersionCommand checks if the given args indicate that the version is being requested.
 func IsVersionCommand(arg string) bool {
 	return isOneOf(arg, VersionArgs)
 }
 
 // PrintVersion prints the cosmovisor version.
-func PrintVersion() error {
+func PrintVersion(logger *zerolog.Logger, args []string) error {
```

Comment:
I don't understand what's going on here tbh. Why are we treating args like this? Isn't cosmovisor executed by cobra? Having the version printed should be super simple.

Reply:
Nope, cosmovisor does not use cobra. I wondered too, but I came across this when I was looking for why: https://github.com/cosmos/cosmos-sdk/pull/10316#issuecomment-939033661

Reply:
Sighhh...Ok then. I guess we can proceed with this, but I must say this approach just feels so...hacky.

Can we at least open up an issue to refactor cosmovisor to use cobra instead? I don't even know why we didn't choose cobra to begin with.

Reply:
Done!